### PR TITLE
Ria 3498 error msg submit

### PR DESCRIPTION
**Pull request by bradsorour**
### JIRA link (if applicable) ###

[RIA-3498](https://tools.hmcts.net/jira/browse/RIA-3498)

### Change description ###

See Jira

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SubmitAppealPreparer.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        if (this.isPaymentAppealTypePayNow(asylumCase)) {
+            asylumCasePreSubmitCallbackResponse.addError("The Submit your appeal option is not available. Select Pay and submit to submit the appeal");
+        }
+
+        return asylumCasePreSubmitCallbackResponse;
+    }
+
+    public boolean isPaymentAppealTypePayNow(AsylumCase asylumCase) {
+        boolean foundPaymentAppealTypePayNow = false;
+        String payNowOption = "payNow";
```
Comment (divisathyan) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SubmitAppealPreparer.java:67]: This could be final or a static constant

Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SubmitAppealPreparer.java]: Made to final

**Review by divisathyan — DISMISSED** [review]
LGTM

**Review by makowalski — CHANGES_REQUESTED** [review]
There is a need to check AIP journey

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SubmitAppealPreparer.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        boolean isRepJourney = callback.getCaseDetails().getCaseData()
+            .read(AsylumCaseFieldDefinition.JOURNEY_TYPE, JourneyType.class)
+            .map(journeyType -> journeyType == JourneyType.REP)
+            .orElse(true);
+
+        return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_START)
+               && Event.SUBMIT_APPEAL == callback.getEvent()
+               && isfeePaymentEnabled
+               && isRepJourney;
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SubmitAppealPreparer.java:48]: Hi @alexbaeza just FYI,
we are adding fees and payments feature check so Rep journey will go through PAY_AND_SUBMIT event
I guess AIP stays as it is within this boolean flag check

**Review by makowalski — APPROVED** [review]

**Pull request merged**